Framework/PCProject.m

```diff
@@ -1345,6 +1345,18 @@ - (BOOL)removeFiles:(NSArray *)files forKey:(NSString *)key notify:(BOOL)yn
 
 - (BOOL)renameFile:(NSString *)fromFile toFile:(NSString *)toFile
 {
+  // to guard from a user error or a bug in text field editors
```

Comment:
Well, in theory, a filename of spaces is a valid filename. Thus, the responsibility should be in the caller - checking the destination name. Controller vs. model. I just tried to implement that. Also, the user should see the original file name restored, if it is rejected. I tried to implement all that, please check.

PS: I prefer having all variable declaration first, then the actual logic.

Reply:
The one of the next commits (9bbe68d) does exactly what you propose. It rejects empty (or only spaces) filename and restores the original file.

A filename of spaces leads to removing the file with the current code. That is what frustrates newcomers. So why it is guarded. It is just a short breaker.

OK, I will abide the rule of  the variable declaration first.